packages/store/test/test-store.js

```diff
@@ -90,37 +83,28 @@ function check(t, mode, objMaker) {
 }
 
 test('store', t => {
-  // makeStore
+  // makeScalarMap
   check(t, 'strong', count => count); // simple numeric keys
   check(t, 'strong', count => `${count}`); // simple strings
   check(t, 'strong', () => Far('handle', {}));
 
-  // makeWeakStore
+  // makeScalarWeakMap
   check(t, 'weak', () => Far('handle', {}));
 });
 
-test('reject unmarked empty objects', t => {
-  // Older client code used harden({}) to create a "handle" that served as an
-  // otherwise-empty key for a store/weakstore, but ticket #2018 changes
-  // marshal to treat unmarked empty objects as pass-by-copy, so they won't
-  // retain identity across messages, breaking old-style handles in
-  // surprising ways (key collisions). New client code should use Far()
-  // instead, which arrives here as an object with a non-empty
-  // getInterfaceOf(). To catch older clients that need to be updated, we
-  // reject the use of plain empty objects as keys.
-
-  const k = harden({});
-  const s = makeStore('store1');
-  t.throws(() => s.init(k, 1), { message: /"store1" bad key:/ });
-  t.throws(() => s.has(k), { message: /"store1" bad key:/ });
-  t.throws(() => s.get(k), { message: /"store1" bad key:/ });
-  t.throws(() => s.set(k, 1), { message: /"store1" bad key:/ });
-  t.throws(() => s.delete(k), { message: /"store1" bad key:/ });
-
-  const w = makeWeakStore('store1');
-  t.throws(() => w.init(k, 1), { message: /"store1" bad key:/ });
-  t.throws(() => w.has(k), { message: /"store1" bad key:/ });
-  t.throws(() => w.get(k), { message: /"store1" bad key:/ });
-  t.throws(() => w.set(k, 1), { message: /"store1" bad key:/ });
-  t.throws(() => w.delete(k), { message: /"store1" bad key:/ });
+test('reject promise keys', t => {
```

Comment:
Do we now accept unmarked hardened empty objects as keys? Is the previous test really now unnecessary?

Reply:
There is no longer anything special about empty objects. A frozen one is just a copyRecord. A scalarMap doesn't accept it as a key just as it doesn't accept any other copyRecord as a key. A scalarMap only accepts primitives and remotables, not containers. A full storeMap will accepts all keys as keys, but will look them up according to sameKey equivalence, which we don't quite have yet. Resolving.